File: Hotel/ClientApp/src/app/_models/transaction.ts
import {jsonIgnore } from 'json-ignore';

export class Transaction {

  constructor() { }
  public transactionId: number;
  public checkInTime: Date;
  public checkOutTime: Date;
  public userId: number;
  public roomId: number;
  public cost: number;
  public isPaid: boolean;
  public isCanceled: boolean;

   // для фильтра
  @jsonIgnore() public UserName: string;
  @jsonIgnore() public ComeIn: string;
  @jsonIgnore() public ComeOut: string;
  @jsonIgnore() public TheNoumber: number;
}

File: Hotel/Data/Entities/Transaction.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using System.ComponentModel.DataAnnotations;

namespace Hotel.Entities
{
    public class Transaction
    {
        public int TransactionId { get; set; }
        [Required, DataType(DataType.Date)]
        public DateTime CheckInTime { get; set; }
        [Required, DataType(DataType.Date)]
        public DateTime CheckOutTime { get; set; }

        [JsonIgnore] public virtual User User { get; set; }
        [JsonIgnore] public virtual Room Room { get; set; }

        [Required]
        public int UserId { get; set; }
        [Required]
        public int RoomId { get; set; }
        [DataType(DataType.Currency)]
        public int Cost { get; set; }
        public bool IsPaid { get; set; }
        public bool IsCanceled { get; set; }
        public bool isEvicted { get; set; }

    }
}
